### PR TITLE
Play card Unit tests

### PR DESCRIPTION
**Pull request by bjsvedin**
I have create a unit test for the playCard function. I also added comments to the play card function, as well as modified some of the checks for better error messages.

### CODE REVIEW

File: server/uno_test.go [hunk 1]
```diff
@@ -18,11 +19,367 @@ func setupGameWithPlayer(database *db.DB) (*model.Game, *model.Player) {
...
+	// Assert that the two players still have all their cards.
+	// Assert that the player turn did not change
+	assert.Equal(t, 1, len(game.DiscardPile))
+	assert.Equal(t, 7, len(game.Players[0].Cards))
+	assert.Equal(t, 7, len(game.Players[1].Cards))
+	assert.Equal(t, 7, len(game.Players[2].Cards))
+	assert.Equal(t, 7, len(game.Players[3].Cards))
+	assert.Equal(t, 0, game.CurrentPlayer)
+
+	game = setUpTestPlayerCard()
```
Comment (logan-hendricks) [inline server/uno_test.go:145]: So. in unit testing... it's absolutely appropriate and more recommended to have each test only test one thing.

In go, that woujld probably look like:

```
TestPlayCard_DiscardSize(t *testing.T) {
    game := setUpTestPlayerCard()
    ...
}

TestPlayCard_HandSize(t *testing.T) {
    game := setUpTestPlayerCard()
    ...
}
... 
etc
```

I would recommend breaking up this larger test into smaller segments of tests, and having each test only focus on 1 aspect of the game. 

BTW, the formatting of the unit test name above is a set convention for Go. One of the recommendations are  `TestPlayCard_${TEST_DESCRIPTION}`. The only important part is the first section `TestPlayCard_`. Whatever comes after is up to you.

Comment (bjsvedin) [reply server/uno_test.go:145]: Sure thing. I can make those changes.

Comment (bjsvedin) [reply server/uno_test.go]: I have split the giant test into separate tests.

**Comment by logan-hendricks** [comment]
@bjsvedin It looks like you have merge conflicts at this point where the logic has changed from dev. I apologize that such is the case, it seems like there's a lot going on right now. Are you available to merge those conflicts? How aggresively are you wanting to get these in?

**Pull request closed without merging**